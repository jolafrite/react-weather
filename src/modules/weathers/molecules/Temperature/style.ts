import { makeStyles, createStyles } from "@material-ui/core";
import { rem } from "polished";

export const useStyles = makeStyles(
  createStyles({
    root: {
      '&.large': {
        '& .text__celcius': {
          fontSize: rem(40),
          lineHeight: rem(42),
        },
        '& .text__separator': {
          fontSize: rem(40),
          lineHeight: rem(40),
        },
        '& .text__fahrenheit': {
          fontSize: rem(38),
          lineHeight: rem(40),
        },
      }
    }
  })
);
