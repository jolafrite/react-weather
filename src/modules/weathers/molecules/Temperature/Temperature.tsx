import React, { FC } from "react";
import { Typography, Grid } from "@material-ui/core";
import { toFahrenheit } from "../../../../common/utils/number";
import clsx from "clsx";
import { useStyles } from "./style";

export interface ITemperatureProps {
  value: number;
  className?: string;
  direction?: "row" | "column";
  separator?: string | null;
  size?: "small" | "large" | "normal";
}

const Temperature: FC<ITemperatureProps> = props => {
  const { className, separator, direction, value, size } = props;
  const classes = useStyles();

  const celcius = value.toFixed(1);
  const fahrenheit = toFahrenheit(value).toFixed(1);

  return (
    <Grid className={clsx(className, size, classes.root)} container direction={direction}>
      <>
        <Typography className="text__celcius">{celcius}Â°C</Typography>
        {separator && (
          <Typography className="text__separator">{separator}</Typography>
        )}
        <Typography className="text__fahrenheit">{fahrenheit}F</Typography>
      </>
    </Grid>
  );
};

Temperature.defaultProps = {
  direction: "row"
};

export default Temperature;
