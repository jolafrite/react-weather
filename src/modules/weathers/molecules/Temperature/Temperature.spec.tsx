import React from 'react';
import { render, cleanup  } from '@testing-library/react';
import Temperature, { ITemperatureProps } from './Temperature';

describe('component: Temperature', () => {

  afterEach(cleanup);

  const createComponent = (values = {}) => {
    const props: ITemperatureProps = {
      value: 30,
      direction: 'row',
      ...values,
    };

    return render(
      <Temperature
        {...props}
      />
    );
  };

  describe('row direction', () => {
    test('should render the Celcius and Fahrenheit on the same line', () => {
      const component = createComponent({
        direction: 'row',
      });

      expect(
        component.container
      ).toMatchSnapshot();
    });
  });



});
