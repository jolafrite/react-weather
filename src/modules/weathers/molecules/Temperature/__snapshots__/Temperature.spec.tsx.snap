// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Temperature row direction should render the Celcius and Fahrenheit on the same line 1`] = `
<div>
  <div
    class="makeStyles-root-1"
    style="flex-direction: row;"
  >
    <p
      class="MuiTypography-root makeStyles-celcius-2 MuiTypography-body1"
    >
      30
      Â°C
    </p>
    <p
      class="MuiTypography-root makeStyles-fahrenheit-4 MuiTypography-body1"
    >
      86
      F
    </p>
  </div>
</div>
`;
